spec/message.abnf

```diff
@@ -1,43 +1,46 @@
-message = pattern / complex-message
+message = simple-message / complex-message
 
-complex-message = "{{" [s] *(declaration [s]) body [s] "}}"
+simple-message = [simple-start pattern]
+simple-start = simple-start-char / text-escape / expression
+pattern = *(text-char / text-escape / expression)
 
-declaration = input-declaration / local-declaration
+complex-message = *(declaration [s]) complex-body
+declaration = input-declaration / local-declaration / reserved-statement
 input-declaration = input [s] variable-expression
 local-declaration = local s variable [s] "=" [s] expression
 
-body = quoted-pattern
-     / (selectors 1*([s] variant))
-
+complex-body = quoted-pattern
+             / ((selectors / reserved-statement) 1*([s] variant))
```

Comment:
Could you say why? The two expressions are wholly synonymous, and I don't really see why we should prefer one over the other.

Reply:
Because I expect the latter to diverge—I don't think we should treat `.match {…}…` and `.<reserved> …` as interchangeable _prefixes_ before a list of variants, but rather `.match {…}… <variant>…` as one thing complete in itself and `.<reserved> …` as something independent.

Reply:
What's the motiviation for putting `reserved-statement` inside `complex-body`? Do we expect other multivariant constructs than `match`? I think this may be building too much flexibility into the spec. Could we instead agree that any future keywords would go with other *declarations*?

Reply:
Tbh, I would find it easier to include it, show what impact it has across the stack, and then evaluate whether to drop it.

We have already agreed on having two follow up PRs to this that will both re-tread this ground. Could we agree that this will be considered more in both of those, and accept having it in the ABNF for now?

Reply:
> Tbh, I would find it easier to include it, show what impact it has across the stack, and then evaluate whether to drop it.

We should do things exactly the other way.

I also don't appreciate pushing hard for merging. We should merge when things look good to merge.